packages/wonder-blocks-icon-button/components/icon-button.js

```diff
@@ -6,14 +6,14 @@ import IconButtonCore from "./icon-button-core.js";
 
 export type SharedProps = {
     /**
-     * An icon, supplied as an svg path string.
+     * The name of a Wonder Blocks icon.
      */
-    icon: string,
+    iconName: string,
 
     /**
      * Text to display as the title of the svg element.
      */
-    alt: string,
+    "aria-label": string,
```

Comment:
Both of these are breaking changes so we should bump the version in its package.json.  Since this hasn't been published yet, bumping the minor version is sufficient.  We'll also need to update `wonder-blocks-modal`'s package.json.  This isn't critical at this point, but I think we should get into the habit of following semver.

Reply:
@kevinbarabash hmm - this feels like a bit of an anti-pattern. We shouldn't be doing version bumps on every commit. Instead, I think it makes more sense to do commits and then do a release that takes into account everything that we're releasing - right? (I guess I'm curious, in your model, how we manage multiple simultaneous small changes to a module. They can't all bump to the same minor version in their commit so we'll end up with merge conflicts.) It seems like it'd be better to land stuff and then do periodic releases of the code that are well publicized and documented.

Reply:
I believe that as long as the current version of `wonder-blocks-icon` satisfies each component that depends on it, those components will use the local version so we could leave things without bumping the versions.  Multiple packages can all bump their minor version if they want.  They'll all be published from the same commit and that commit will have multiple tags, e.g. `@khanacademy/wonder-blocks-foo@0.0.2`, `@khanacademy/wonder-blocks-bar@0.02` are what the tags would look like.  I don't think we should publish for every version bump, but if we don't bump the version in the commit then we have to remember to bump it later.  I also worry about inconsistencies where something says it uses version `0.1.0` but actually needs `0.2.0`.

Reply:
@kevinbarabash Interesting, so I think there are two separate issues: Ensuring that our published/released code points to the right versions and ensuring that the code inside Github represents reality.

What if we did something like this:

- Write some code, submit as PR, land code.
- Publish module "foo" 0.0.1 -> 0.0.2, which is depended upon by "bar" so we bump that from 0.0.1 -> 0.0.2, as well.
- Immediately (as part of the publish process) we add a commit that changes the versions of these modules as something like 0.0.3-pre and leave them in that state until the next publish (at which point we change the version from 0.0.3-pre to 0.0.3 or 0.1.0 or whatever).

Alternatively, we could add a pre-commit hook and have it auto-bump the version to "0.0.3-pre" if a package has any code changed in it. That will still have the issue with merge conflicts, but at least it won't be as bad as they'll be attempting to go to the same version, rather than potentially different ones.

This way code inside Github, that other devs see, will always represent a WIP version of the component. If they want to see code at a specific release then they need to view the version tag. I guess the downside with this is that the version of what we publish to Netlify won't, necessarily, match what users can actually use -- but that's also true even if we're doing publishing (e.g. something that we publish to NPM won't be immediately available in webapp). We may want to consider having a branch or tag inside `wonder-blocks` that points to what we're currently using on webapp -- and then only publish that to the main Netlify (we could publish the WIP stuff to something like wonder-blocks-dev.netlify.com instead).

Reply:
We talked about this IRL and decided that devs won't be responsible for version bumping.  The specifics of how version bumping will happen will be documented somewhere in a Google doc.